[qtbase] Fix Qt6 for MacOS

eschablowski opened the pull request:
Fixes https://github.com/microsoft/vcpkg/issues/41710
- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

> I am unsure if the fix to support MacOS 15 in the qtbase and qtmultimedia fully complies with [Prefer patching over overriding VCPKG_<VARIABLE> values](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#prefer-patching-over-overriding-vcpkg_variable-values) or if it should use [Recommended Solutions](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#recommended-solutions)

eschablowski committed:
Fix Qt on MacOS 15

Signed-off-by: Elias Schablowski <eschablowski@me.com>

eschablowski committed:
Fix qt dependencies for MacOS

Signed-off-by: Elias Schablowski <eschablowski@me.com>

jimwang118 commented:
You should modify the [port-version](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#update-the-port-version-field-in-the-manifest-file-of-any-modified-ports).

eschablowski committed:
Add explicit port versions for qt, qtbase, and qtmultimedia

Signed-off-by: Elias Schablowski <eschablowski@me.com>

eschablowski committed:
Add versions from the `vcpkg x-add-version --all` command

Signed-off-by: Elias Schablowski <eschablowski@me.com>

eschablowski commented:
> You should modify the [port-version](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#update-the-port-version-field-in-the-manifest-file-of-any-modified-ports).

Thank you for pointing this out, I just fixed this, apparently I flew over the guidelines a bit too hastily.
I also added the missing newline while I was in the manifest.

kikaxa commented on ports/qtbase/portfile.cmake:
```diff
@@ -43,6 +43,17 @@ if(VCPKG_TARGET_IS_LINUX)
+if(VCPKG_TARGET_IS_OSX AND (VCPKG_OSX_DEPLOYMENT_TARGET VERSION_GREATER_EQUAL "15.0") )
+    message(WARNING "Qt6 does not yet support macOS 15.0, building Qt6 for MacOS 14.")
+    set(VCPKG_OSX_DEPLOYMENT_TARGET "14.0")
+endif()
```
there are many more modules that dont have this fix added.
imo they should have this same version to be consistent in safe on any abi/headers mismatch that can happen.
perhaps cmake/qt_install_submodule.cmake is a better place?

m-kuhn replied:
For consistency, should this be globally set in the triplet?
i.e. only show a warning message here suggesting to set this in the triplet file (and remove line 54)

Neumann-A replied:
does the warning still apply to qt 6.8 ?

Neumann-A replied:
My advise is to set 

> If not set explicitly the value is initialized by the MACOSX_DEPLOYMENT_TARGET environment variable, if set, and otherwise computed based on the host platform.

in your build environment.

BillyONeal replied:
> does the warning still apply to qt 6.8 ?

As far as I am aware it does. In macOS 15 / XCode 16, Apple just outright removed APIs that qt uses. https://github.com/microsoft/vcpkg/pull/41307

xavier2k6 commented:
Qt 6.8.0 has support for macOS 15.0 (Sequoia)

eschablowski commented:
This issue is unfortunately not fixed in Qt 6.8, even though Qt added official support for Sequoia: https://bugreports.qt.io/browse/QTBUG-128900, and the line is still unchanged: https://code.qt.io/cgit/qt/qtbase.git/tree/src/plugins/platforms/cocoa/qcocoascreen.mm#n550

eschablowski committed:
Stop auto-setting, in favor of suggestion system

Signed-off-by: Elias Schablowski <eschablowski@me.com>

eschablowski committed:
Remove extainious newline

eschablowski committed:
Update version commits

eschablowski commented:
As the general consensus was to have a suggestion to use a custom triplet, the defaulting behavior has been removed in favor of a suggestion mechanism.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

# Conflicts:
#	ports/qt/vcpkg.json
#	ports/qtbase/vcpkg.json
#	ports/qtmultimedia/vcpkg.json
#	versions/baseline.json
#	versions/q-/qt.json
#	versions/q-/qtbase.json
#	versions/q-/qtmultimedia.json

BillyONeal committed:
Fix the comment for 6.8.
